query-engine/nodejs-drivers/nodejs-examples/src/queryable/mysql.ts

```diff
@@ -41,33 +129,24 @@ class PrismaMySQL implements Queryable, Closeable {
   /**
    * Execute a query given as SQL, interpolating the given parameters.
    */
-  async queryRaw(query: string): Promise<ResultSet> {
-    console.log('[nodejs] calling queryRaw', query)
+  async queryRaw(params: Query): Promise<ResultSet> {
+    console.log('[nodejs] calling queryRaw', params)
     const [results, fields] = await this.pool.query<mysql.RowDataPacket[]>({
-      sql: query,
+      sql: params.sql,
+      values: params.args,
       rowsAsArray: false,
     })
     console.log('[nodejs] after query')
 
     const columns = fields.map(field => field.name)
     const resultSet: ResultSet = {
-      columns: columns,
-
-      // TODO: what if I remove the `toString()`?
-      // Currently, it would fail with something like:
-      // [Error: Failed to convert JavaScript value `Number 1 ` into rust type `String`],
-      // because the `id` column is a number, but the `ResultSet` expects a Vec<Vec<Str>>.
-      rows: results.map(result => columns.map(column => result[column].toString()))
+      columnNames: columns,
+      // TODO: cancel the promise and fail in case the column type is not supported by the driver
+      columnTypes: fields.map(field => fieldToColumnType(field)),
```

Comment:
We can talk about this comment in tomorrow's call, I'm eager to better understand what you mean here